core/src/main/java/com/neemre/btcdcli4j/core/client/BtcdClient.java

```diff
@@ -75,7 +75,31 @@ RedeemScript decodeScript(String hexRedeemScript) throws BitcoindException,
 
 	String encryptWallet(String passphrase) throws BitcoindException, CommunicationException;
 
-	BigDecimal estimateFee(Integer maxBlocks) throws BitcoindException, CommunicationException;
+	/**
+	 * See https://developer.bitcoin.org/reference/rpc/estimatesmartfee.html
+	 * @param maxBlocks Confirmation target in blocks (1 - 1008)
+	 * @param estimateMode optional, default=CONSERVATIVE
```

Comment:
I think it should not be optional here in this case imo. It should be mandatory. User should use the overloaded method if they do not want to provide it.

Reply:
@kushal256 halp! I can't really tell where the default value takes place of null argument. Is it done by the external service called by gRPC? I mean... is it like we provide null to that external service and then the external service is the one which assumes conservative?

Reply:
> is it like we provide null to that external service and then the external service is the one which assumes conservative?

Exactly, as noted in the link in the javadoc

Reply:
> > is it like we provide null to that external service and then the external service is the one which assumes conservative?
> 
> Exactly, as noted in the link in the javadoc

My bad 😅 glad I didn't hold you any further because of that question 😂

Reply:
More questions are always good!  Perhaps that means I should have noted that in the javadoc??  WDYT?